### ISSUE TITLE
[iceprod2] fix pilot resource matching (Trac #1694)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1686]
Currently, a pilot will just grab the next job to run, without bothering to match resources. Let's fix the `# TODO: add resources like GPUs, high memory, etc`.

This is kind-of required if we want to run more than one task at a time.

Migrated from https://code.icecube.wisc.edu/ticket/1694
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:28", 
    "description": "Currently, a pilot will just grab the next job to run, without bothering to match resources. Let's fix the `# TODO: add resources like GPUs, high memory, etc`.\n\nThis is kind-of required if we want to run more than one task at a time.", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067088921308", 
    "component": "iceprod", 
    "summary": "[iceprod2] fix pilot resource matching", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2016-05-05T20:02:10", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160506T15:31:04: 
* david.schultz changed resolution from "" to "fixed"
* david.schultz changed status from "new" to "closed"
* david.schultz commented:

> In changeset 145554:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="145554"
> pilot resource matching, and some other resource fixes. closes #1686
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:28: 
* olivas changed milestone from "IceProd 2.0" to ""
* olivas commented:

> Milestone IceProd 2.0 deleted
